environment/src/main/java/com/sequenceiq/environment/environment/validation/cloudstorage/CloudStorageLocationValidator.java

```diff
@@ -0,0 +1,50 @@
+package com.sequenceiq.environment.environment.validation.cloudstorage;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.providerservices.CloudProviderServicesV4Endopint;
+import com.sequenceiq.cloudbreak.client.CloudbreakServiceUserCrnClient;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.cloud.model.objectstorage.ObjectStorageMetadataRequest;
+import com.sequenceiq.cloudbreak.cloud.model.objectstorage.ObjectStorageMetadataResponse;
+import com.sequenceiq.cloudbreak.util.ValidationResult.ValidationResultBuilder;
+import com.sequenceiq.environment.credential.v1.converter.CredentialToCloudCredentialConverter;
+import com.sequenceiq.environment.environment.domain.Environment;
+
+@Component
+public class CloudStorageLocationValidator {
+
+    @Inject
+    private CloudbreakServiceUserCrnClient cloudbreakClient;
```

Comment:
please use ctor injection in env service

Reply:
nope

Reply:
do you have any strong objection against ctor injection in this case? if not I would ask for it also

Reply:
yes, actually I prefer explicitly expressing a framework feature (injection) rather than relying on injection by conventions

Reply:
even the authors of the framework say developers should use the ctor injection, because it's the better practice

Reply:
see above